packages/react-aria-components/src/utils.tsx

```diff
@@ -326,3 +326,20 @@ export function createHideableComponent<T, P = {}>(fn: (props: P, ref: React.Ref
   Wrapper.displayName = fn.displayName || fn.name;
   return (React.forwardRef as forwardRefType)(Wrapper);
 }
+
+/**
+ * Filters out `data-*` attributes to keep them from being passed down and duplicated.
+ * @param props
+ */
+export function filterDataAttrs(props: SharedDOMProps & AriaLabelingProps): SharedDOMProps & AriaLabelingProps {
```

Comment:
the name of this function is opposite to filterDOMProps which filters to *keep* dom props
can we name this one `filterOutDataAttributes`? or something that won't cause me to mix up the two as I did while I was reviewing just a moment ago :)